compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -295,6 +297,24 @@ public void visit(BUnionType bUnionType) {
         for (BType memberType : bUnionType.getMemberTypes()) {
             writeTypeCpIndex(memberType);
         }
+
+        if (tsymbol instanceof BEnumSymbol) {
+            buff.writeBoolean(true);
+            writeEnumSymbolInfo((BEnumSymbol) tsymbol);
+        } else {
+            buff.writeBoolean(false);
+        }
```

Comment:
@Kishanthan I'm changing the format of the binary a bit here. Do we need to update the kaitai file as well?

Reply:
@pubudu91 yes :)